server/controllers/conversation.js

```diff
@@ -0,0 +1,108 @@
+const mongoose = require("mongoose");
+const Conversation = require("../models/Conversation");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /conversations
+// @desc Get all conversations of a logged-in user
+// @access private
+exports.getAllConversations = asyncHandler(async (req, res, next) => {
+  try {
+    const userId = req.user.id;
+
+    const conversations = await Conversation.find({
+      $or: [{ user1: userId }, { user2: userId }],
+    })
+      .populate("messages")
+      .populate("user1", "-password -register_date -__v")
+      .populate("user2", "-password -register_date -__v");
+
+    res.status(200).json(conversations);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// @route GET /conversations/:conversationId
+// @desc Get a conversation (all messages in the conversation) of a logged-in user
+// @access private
+exports.getConversation = asyncHandler(async (req, res, next) => {
+  try {
+    const { conversationId } = req.params;
+
+    if (!mongoose.Types.ObjectId.isValid(conversationId)) {
+      res.status(400);
+      throw new Error("Invalid conversationId");
+    }
+
+    const userId = req.user.id;
+
+    let conversation;
+
+    conversation = await Conversation.findOne({
+      _id: conversationId,
+      $or: [{ user1: userId }, { user2: userId }],
+    });
+
+    if (!conversation) {
+      res.status(404);
+      throw new Error(
+        "Couldn't find a conversation for the user with the conversationId"
+      );
+    }
+
+    conversation = await Conversation.findById(conversationId)
```

Comment:
You  have already have the required data in conversation, May I know why you queried it again ?

Reply:
I wanted to populate related `messages`, `user1`, and `user2` from a `conversation`. So I queried it again.
However, after I read your feedback, I think it's not necessary. (`findById`)

As you mentioned, I already have the required data in conversation. So I chained `.populate()` with the first query(`findOne`), and removed unnecessary query(`findById`). The result of this route is the same.
Thank you for your feedback. 🙂